### PR TITLE
tx version break step with server version

### PR DESCRIPTION
**Pull request by libscott**
Transaction now has its own version and it must be `1.0` to validate.

Part of the resolution of #1558

### CODE REVIEW

File: tests/validation/test_transaction_structure.py [hunk 1]
```diff
@@ -165,14 +165,7 @@ def test_high_amounts(create_tx):
...
-    short_ver = bigchaindb.version.__short_version__
-    assert create_tx.version == re.match(r'^(.*\d)', short_ver).group(1)
-
-    validate(create_tx)
-
-    # At version 1, transaction version will break step with server version.
-    create_tx.version = '1.0.0'
+    create_tx.version = '0.10'
+    validate_raises(create_tx)
+    create_tx.version = '110'
```
Comment (ttmc) [inline tests/validation/test_transaction_structure.py]: What about a test to see if `'1.0'` is okay?

File: bigchaindb/common/transaction.py [hunk 2]
```diff
@@ -421,7 +420,7 @@ class Transaction(object):
     TRANSFER = 'TRANSFER'
     GENESIS = 'GENESIS'
     ALLOWED_OPERATIONS = (CREATE, TRANSFER, GENESIS)
-    VERSION = '.'.join(bigchaindb.version.__short_version__.split('.')[:2])
+    VERSION = '1.0'
```
Comment (r-marques) [inline bigchaindb/common/transaction.py]: Why do we need to change this. Wouldn't be enough to update `version.py`?

File: bigchaindb/common/transaction.py [hunk 3]
```diff
@@ -441,7 +440,6 @@ def __init__(self, operation, asset, inputs=None, outputs=None,
                     lock.
                 metadata (dict): Metadata to be stored along with the
                     Transaction.
-                version (int): Defines the version number of a Transaction.
```
Comment (r-marques) [inline bigchaindb/common/transaction.py]: This should not be removed but we could change the type of version to `str`

File: tests/validation/test_transaction_structure.py (hunk 1 above)
Comment (libscott) [inline tests/validation/test_transaction_structure.py]: implied by all the other tests

File: bigchaindb/common/transaction.py (hunk 2 above)
Comment (libscott) [inline bigchaindb/common/transaction.py]: Transaction version is breaking step with server version, so it makes sense to hardcode it in the code that generates the transactions. When the server version changes, the transaction version should not neccesarily change.

File: bigchaindb/common/transaction.py (hunk 3 above)
Comment (libscott) [inline bigchaindb/common/transaction.py]: Oops

File: tests/validation/test_transaction_structure.py (hunk 1 above)
Comment (libscott) [inline tests/validation/test_transaction_structure.py]: Added by popular demand

File: bigchaindb/common/transaction.py (hunk 2 above)
Comment (r-marques) [inline bigchaindb/common/transaction.py]: Up until now the transactions had the same version as the server.
This is saying that the transaction version and the server version should be two different things.
I don't necessarily disagree but this is a decision that affects all bigchaindb drivers

Comment (ttmc) [inline bigchaindb/common/transaction.py]: Tim made a TODO list of all the things that must be done, including updating all the drivers: https://github.com/bigchaindb/bigchaindb/issues/1558#issuecomment-310317072

**Pull request merged**